### DISCUSSION TITLE
->pdf(); nothing happens?

### DISCUSSION THREAD

**Discussion by yankeyhotel** [spatie/browsershot#506] [Q&A]
I am having trouble creating and then immediately downloading a PDF.

My expectation is that this would result in the pdf downloading from the browser, but nothing happens. (https://github.com/spatie/browsershot#output-directly-to-the-browser-1)
```
Browsershot::html('<h1>Hello</h1>')
    ->setNodeBinary('/usr/local/bin/node')
    ->setNpmBinary('/usr/local/bin/npm')
    ->format('Letter')
    ->pdf();
```

This code works just fine, the example.pdf is created and added to the root.
```
Browsershot::html('<h1>Hello</h1>')
    ->setNodeBinary('/usr/local/bin/node')
    ->setNpmBinary('/usr/local/bin/npm')
    ->format('Letter')
    ->save('example.pdf');
```

`->screenshot()` doesn't work either. (https://github.com/spatie/browsershot#output-directly-to-the-browser)

Thanks

**Comment by stefanzweifel** [comment]
`pdf()` returns the PDF as a string ([Source](https://github.com/spatie/browsershot/blob/master/src/Browsershot.php#L558-L567)).

If the PDF should be downloaded from a controller, you have to transform it into the appropriate response:

```php

$pdf = Browsershot::html('<h1>Hello</h1>')
    ->setNodeBinary('/usr/local/bin/node')
    ->setNpmBinary('/usr/local/bin/npm')
    ->format('Letter')
    ->pdf();

return \Response::make($pdf, 200, ['Content-Type' => 'application/pdf']);
```
